code/modules/projectiles/gun.dm

```diff
@@ -52,6 +52,7 @@
 	var/tmp/list/mob/living/target //List of who yer targeting.
 	var/tmp/lock_time = -100
 	var/mouthshoot = 0 ///To stop people from suiciding twice... >.>
+	var/canceled = 0 //flags if a suicide was canceled
```

Comment:
this seems like a really weird implementation

wouldn't everything work fine if you just made the cell drain part of process_chambered() instead of attack()? I believe energy guns call this too

Reply:
It felt weird when I wrote it, but I was at a loss for alternatives. Good tip about process_chambered(), I didn't think energy guns called it for some reason, but you're right.